prov/rxm/src/rxm_cq.c

```diff
@@ -1237,7 +1237,10 @@ int rxm_ep_prepost_buf(struct rxm_ep *rxm_ep, struct fid_ep *msg_ep)
 	int ret;
 	size_t i;
 
-	for (i = 0; i < rxm_ep->msg_info->rx_attr->size; i++) {
+	/* -1 because we always post the next recv before releasing the previous
+	 * one. Without the -1 we immediately exceed the starting recv pool size.
+	 */
```

Comment:
As I mentioned in #4822 , we post new recv without releasing previous one only when there are unexpected messages. There is no need for -1 in regular case. Do you see a case where we exceed the starting recv pool size?

Reply:
Chris' commented that the post occurs prior to the receive being released, which results in the +1.

Reply:
I checked the rxm, and @a-ilango is correct.  A received message is not automatically re-posted.  The receive is processed, including copying out any data, prior to the receive being placed on a re-post list.  The receive is re-posted to the MSG EP from that list as part of the progress call.  I will drop this patch and close #4822.